File: DTC.Api/Dtos/Braket/BraketDto.cs
using DTC.Api.Dtos.Match;
using DTC.Api.Enums;

namespace DTC.Api.Dtos.Braket
{
    public class BraketDto
    {
        public int Id { get; set; }

        public int TournamentId { get; set; }

        public int Sequence { get; set; }

        public string? Name { get; set; }

        public DateTimeOffset PlannedStart { get; set; }

        public DateTimeOffset? PlannedEnd { get; set; }

        public BraketStatus Status { get; set; }

        public BraketPhase Phase { get; set; }

    }
}

File: dtc.frontend/src/dtos/brakets/BraketDto.ts
import z from "zod";
import { BraketStatus } from "../../enums/BraketStatus";
import { BraketPhase } from "../../enums/BraketPhase";

export const braketDtoSchema = z.object({
  id: z.number().int(),
  tournamentId: z.number().int(),
  sequence: z.number().int(),
  name: z.string().max(100).nullable(),
  plannedStart: z.iso.datetime({ offset: true }),
  plannedEnd: z.iso.datetime({ offset: true }).nullable(),
  status: z.enum(BraketStatus),
  phase: z.enum(BraketPhase),
});

export type BraketDto = z.infer<typeof braketDtoSchema>;
